### Homework 2

### Question 1: Comparing Two Population Means (Independent t-test)

### Hypotheses

- Null hypothesis ($H_0$): The mean increase in sleep is the same for both drugs, $\mu_1 - \mu_2 = 0$.
- Alternative hypothesis ($H_A$): The mean increase in sleep differs between the drugs, $\mu_1 - \mu_2 \ne 0$.

Because the question asks whether the means differ, this is a two-sided independent-samples t-test. Welch's version is used because it does not assume equal population variances.

The significance level is $\alpha = 0.05$.

In [ ]:
import numpy as np
from scipy import stats


sleep_drug_1 = np.array([0.7, -1.6, -0.2, -1.2, -0.1, 3.4, 3.7, 0.8, 0.0, 2.0])
sleep_drug_2 = np.array([1.9, 0.8, 1.1, 0.1, -0.1, 4.4, 5.5, 1.6, 4.6, 3.4])

alpha = 0.05
result = stats.ttest_ind(sleep_drug_1, sleep_drug_2, equal_var=False)
mean_difference = sleep_drug_1.mean() - sleep_drug_2.mean()

print(f"Drug 1 mean: {sleep_drug_1.mean():.2f} hours")
print(f"Drug 2 mean: {sleep_drug_2.mean():.2f} hours")
print(f"Mean difference (drug 1 - drug 2): {mean_difference:.2f} hours")
print(f"Welch t-statistic: {result.statistic:.3f}")
print(f"Two-sided p-value: {result.pvalue:.4f}")

if result.pvalue < alpha:
    print("Decision: Reject H0. The mean increases differ significantly.")
else:
    print("Decision: Fail to reject H0. There is not enough evidence of a significant difference.")

Drug 1 mean: 0.75 hours
Drug 2 mean: 2.33 hours
Mean difference (drug 1 - drug 2): -1.58 hours
Welch t-statistic: -1.861
Two-sided p-value: 0.0794
Decision: Fail to reject H0. There is not enough evidence of a significant difference.


### Results and conclusion

- The sample mean increase was 0.75 hours for drug 1 and 2.33 hours for drug 2. 
- Thus, drug 2 had a sample mean increase that was 1.58 hours larger than drug 1.

The Welch independent t-test produced $t = -1.861$ and a two-sided $p$-value of 0.0794. Since $p > 0.05$, we fail to reject $H_0$. There is not sufficient statistical evidence at the 5% significance level to conclude that the mean increase in sleep differs between the two drugs.

This conclusion does not prove that the drugs have identical effects; it means that this sample does not provide strong enough evidence of a difference at $\alpha = 0.05$.

### Question 2 ANOVA 
The iris dataset contains measurements for 50 flowers from each of three species of iris: 
setosa, versicolor, and virginica. 

### Context: 
- The iris dataset contains measurements for 50 flowers from each of three species of iris:setosa, versicolor, and virginica. 

### Question 2: 
- Is there a statistically significant difference in the mean sepal length (Sepal.Length) among the three different iris species (Species)? 
- If so, which groups differ?

In [6]:
import pandas as pd 
import numpy as np 
import seaborn as sns


iris_data = sns.load_dataset('iris')

In [7]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

alpha = 0.05

# One-way ANOVA: compare mean sepal length across the three species.
groups = [
    iris_data.loc[iris_data["species"] == species, "sepal_length"]
    for species in iris_data["species"].unique()
]
anova_result = stats.f_oneway(*groups)

print("Mean sepal length by species:")
print(iris_data.groupby("species")["sepal_length"].mean().round(3))
print(f"\nANOVA F-statistic: {anova_result.statistic:.3f}")
print(f"ANOVA p-value: {anova_result.pvalue:.6g}")

if anova_result.pvalue < alpha:
    print("\nDecision: Reject H0. At least one species mean differs.")

    tukey_result = pairwise_tukeyhsd(
        endog=iris_data["sepal_length"],
        groups=iris_data["species"],
        alpha=alpha,
    )
    print("\nTukey HSD post-hoc comparisons:")
    print(tukey_result)
else:
    print("\nDecision: Fail to reject H0. There is not enough evidence that the means differ.")

Mean sepal length by species:
species
setosa        5.006
versicolor    5.936
virginica     6.588
Name: sepal_length, dtype: float64

ANOVA F-statistic: 119.265
ANOVA p-value: 1.66967e-31

Decision: Reject H0. At least one species mean differs.

Tukey HSD post-hoc comparisons:
   Multiple Comparison of Means - Tukey HSD, FWER=0.05   
  group1     group2   meandiff p-adj lower  upper  reject
---------------------------------------------------------
    setosa versicolor     0.93   0.0 0.6862 1.1738   True
    setosa  virginica    1.582   0.0 1.3382 1.8258   True
versicolor  virginica    0.652   0.0 0.4082 0.8958   True
---------------------------------------------------------


### Results and conclusion

The mean sepal lengths are:

- **Setosa:** 5.006
- **Versicolor:** 5.936
- **Virginica:** 6.588

The one-way ANOVA produced F(2,147) = 119.265 with $p \approx 1.67 \times 10^{-31}$. Because $p < 0.05$, we reject $H_0$ and conclude that mean sepal length is not the same for all three iris species.

Tukey's HSD post-hoc test found that every pair differs significantly at the 5% level:

- Setosa has a significantly shorter mean sepal length than versicolor.
- Setosa has a significantly shorter mean sepal length than virginica.
- Versicolor has a significantly shorter mean sepal length than virginica.

Therefore, all three species differ significantly in mean sepal length, with virginica having the largest sample mean and setosa the smallest.

### Question 3 Chi-Square Test for Independence 


### Context: 
- The mtcars dataset gives performance data for 32 different cars. The variable cyl 
represents the number of cylinders (4, 6, or 8) and am represents the transmission type (0 = 
automatic, 1 = manual).  

### Question 3: 
- Is there a statistically significant association between the number of cylinders a car has 
and its transmission type? 

In [ ]:
import pandas as pd
from scipy import stats 

# Chi-Square Test for Independence between number of cylinders and transmission type in the mtcars dataset

cylinders = [6, 6, 4, 6, 8, 6, 8, 4, 4, 6, 6, 8, 8, 8, 8, 8,
            8, 4, 4, 4, 4, 8, 8, 8, 8, 4, 4, 4, 8, 8, 4]
transmission = [1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1]

mtcars = pd.DataFrame({"cyl": cylinders, "am": transmission})
contingency_table = pd.crosstab(mtcars["cyl"], mtcars["am"])
chi2_result = stats.chi2_contingency(contingency_table)

print("Contingency table (rows = cylinders, columns = transmission):")
print(contingency_table.rename(columns={0: "automatic", 1: "manual"}))
print("\nExpected counts:")
print(pd.DataFrame(
    chi2_result.expected_freq,
    index=contingency_table.index,
    columns=["automatic", "manual"],
).round(2))
print(f"\nChi-square statistic: {chi2_result.statistic:.3f}")
print(f"Degrees of freedom: {chi2_result.dof}")
print(f"p-value: {chi2_result.pvalue:.6f}")

if chi2_result.pvalue < 0.05:
    print("\nDecision: Reject H0. Cylinder count and transmission type are associated.")
else:
    print("\nDecision: Fail to reject H0. There is not enough evidence of an association.")

Contingency table (rows = cylinders, columns = transmission):
am   automatic  manual
cyl                   
4            3       8
6            4       2
8           12       2

Expected counts:
     automatic  manual
cyl                   
4         6.74    4.26
6         3.68    2.32
8         8.58    5.42

Chi-square statistic: 8.958
Degrees of freedom: 2
p-value: 0.011343

Decision: Reject H0. Cylinder count and transmission type are associated.


### Results and conclusion

- Null hypothesis ($H_0$): The number of cylinders and transmission type are independent.
- Alternative hypothesis ($H_A$): The number of cylinders and transmission type are associated.

The contingency table is:

| Cylinders | Automatic | Manual |
|---:|---:|---:|
| 4 | 3 | 8 |
| 6 | 4 | 2 |
| 8 | 12 | 2 |

The chi-square test produced $\chi^2(2) = 8.958$ with a $p$-value of 0.011343. Since $p < 0.05$, we reject $H_0$ and conclude that the number of cylinders and transmission type are statistically significantly associated.

In this sample, manual transmissions are more common among 4-cylinder cars, while automatic transmissions are especially common among 8-cylinder cars.

### Question 4 Chi-Square Goodness-of-Fit Test 


### Context: 
- An economist claims that among commuters in a large city, the preferred mode of 
transport is distributed as follows: 50% use public transit, 30% drive their own car, and 20% use 
other methods (cycling, walking, etc.). A random sample of 300 commuters is taken, and their 
choices are recorded: 135 use public transit, 95 drive their own car, and 70 use other methods.   

### Question 4: 
-  At the alpha=0.01 significance level, does the sample data provide evidence to reject the 
economist's claim about the distribution of commuter transport preferences? 

In [12]:
import numpy as np
from scipy import stats

observed_counts = np.array([135, 95, 70])
claimed_proportions = np.array([0.50, 0.30, 0.20])
sample_size = observed_counts.sum()
expected_counts = sample_size * claimed_proportions
alpha = 0.01

gof_result = stats.chisquare(
    f_obs=observed_counts,
    f_exp=expected_counts,
)

categories = ["Public transit", "Own car", "Other methods"]
print("Observed counts:", dict(zip(categories, observed_counts)))
print("Expected counts:", dict(zip(categories, expected_counts)))
print(f"\nChi-square statistic: {gof_result.statistic:.3f}")
print(f"Degrees of freedom: {len(observed_counts) - 1}")
print(f"p-value: {gof_result.pvalue:.6f}")

if gof_result.pvalue < alpha:
    print("\nDecision: Reject H0. The sample distribution differs from the claimed distribution.")
else:
    print("\nDecision: Fail to reject H0. There is not enough evidence to reject the claimed distribution.")

Observed counts: {'Public transit': 135, 'Own car': 95, 'Other methods': 70}
Expected counts: {'Public transit': 150.0, 'Own car': 90.0, 'Other methods': 60.0}

Chi-square statistic: 3.444
Degrees of freedom: 2
p-value: 0.178669

Decision: Fail to reject H0. There is not enough evidence to reject the claimed distribution.


### Results and conclusion

- Null hypothesis ($H_0$): The commuter preferences follow the claimed distribution $(0.50, 0.30, 0.20)$.
- Alternative hypothesis ($H_A$): The commuter preferences do not follow the claimed distribution.

For a sample of 300 commuters, the expected counts are 150 public transit users, 90 own-car drivers, and 60 users of other methods. The observed counts are 135, 95, and 70, respectively.

The chi-square goodness-of-fit test produced $\chi^2(2) = 3.444$ with a $p$-value of **0.178669**. Because $p > \alpha=0.01$, we **fail to reject $H_0$**.

Therefore, at the 1% significance level, the sample does not provide sufficient evidence to reject the economist's claim about the distribution of commuter transport preferences.

### Question 5 Nonparametric Testing 


### Context: 
- Revisit the sleep dataset from Question 1. One key assumption of the t-test is that the data 
for each group is approximately normally distributed, especially with small sample sizes (n=10 for 
each group).   

### Question 5: 
-  If we are unwilling to assume normality for the extra sleep data, what is an appropriate 
nonparametric alternative to the t-test? Perform this test and compare your conclusion to the one 
from Question 1. 

In [13]:
mann_whitney_result = stats.mannwhitneyu(
    sleep_drug_1,
    sleep_drug_2,
    alternative="two-sided",
)

print(f"Drug 1 median: {np.median(sleep_drug_1):.2f} hours")
print(f"Drug 2 median: {np.median(sleep_drug_2):.2f} hours")
print(f"Mann-Whitney U statistic: {mann_whitney_result.statistic:.1f}")
print(f"Two-sided p-value: {mann_whitney_result.pvalue:.4f}")

if mann_whitney_result.pvalue < 0.05:
    print("Decision: Reject H0. The two groups differ significantly.")
else:
    print("Decision: Fail to reject H0. There is not enough evidence of a significant difference.")

Drug 1 median: 0.35 hours
Drug 2 median: 1.75 hours
Mann-Whitney U statistic: 25.5
Two-sided p-value: 0.0693
Decision: Fail to reject H0. There is not enough evidence of a significant difference.


### Results and comparison with Question 1

The appropriate nonparametric alternative to the independent-samples t-test is the **two-sided Mann-Whitney U test**. It does not require normally distributed data and compares the rank distributions of the two independent groups.

- Null hypothesis ($H_0$): The two groups come from the same distribution.
- Alternative hypothesis ($H_A$): The two groups come from different distributions.

The median increase was 0.35 hours for drug 1 and 1.75 hours for drug 2. The Mann-Whitney test produced $U=25.5$ with a two-sided $p$-value of **0.0693**. Since $p>0.05$, we fail to reject $H_0$.

This agrees with Question 1: the Welch t-test also failed to find a statistically significant difference ($p=0.0794$). Therefore, whether we compare means with the t-test or rank distributions with the Mann-Whitney test, this sample does not provide sufficient evidence of a difference between the drugs at the 5% significance level.

### Question 6 Comprehensive Regression Modeling and Multicollinearity 


### Context: 
- An economist wants to build the best possible model to explain and predict a car's fuel 
efficiency (mpg) using the mtcars dataset. The potential predictors include weight (wt), 
horsepower (hp), number of cylinders (cyl), and transmission type (am).  

### Question 6: 
-  a) Begin by fitting a "full" model that includes wt, hp, cyl, and am as predictors of mpg. Interpret the 
coefficient for hp. 

- b) OLS Assumption: No Perfect Multicollinearity. In a model with wt, hp, and cyl, high correlation 
between predictors (multicollinearity) can be a problem. Calculate the Variance Inflation Factor 
(VIF) for these predictors. Based on the VIF values and the results from part (a), build and justify a 
more parsimonious (simpler) model. 

- c) Interaction Effects. Using your simpler model from part (b), test if the effect of weight (wt) on 
mpg depends on the transmission type (am). Fit a new model including this interaction term. 
Interpret the coefficient of the interaction term. 

- d) Based on your analyses, which model do you believe is "best"? Justify your choice by comparing 
the models on interpretability, adjusted R-squared, and statistical significance. 

In [ ]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.stats.outliers_influence import variance_inflation_factor

mtcars = pd.DataFrame({
    "mpg": [21.0, 21.0, 22.8, 21.4, 18.7, 18.1, 14.3, 24.4, 22.8, 19.2,
            17.8, 16.4, 17.3, 15.2, 10.4, 10.4, 14.7, 32.4, 30.4, 33.9,
            21.5, 15.5, 15.2, 13.3, 19.2, 27.3, 26.0, 30.4, 15.8, 19.7,
            15.0, 21.4],
    "wt": [2.620, 2.875, 2.320, 3.215, 3.440, 3.460, 3.570, 3.190,
            3.150, 3.440, 3.440, 4.070, 3.730, 3.780, 5.250, 5.424,
            5.345, 2.200, 1.615, 1.835, 2.465, 3.520, 3.435, 3.840,
            3.845, 1.935, 2.140, 1.513, 3.170, 2.770, 3.570, 2.780],
    "hp": [110, 110, 93, 110, 175, 105, 245, 62, 95, 123, 123, 180,
            180, 180, 205, 215, 230, 66, 52, 65, 97, 150, 150, 245,
            175, 66, 91, 113, 264, 175, 335, 109],
    "cyl": [6, 6, 4, 6, 8, 6, 8, 4, 4, 6, 6, 8, 8, 8, 8, 8,
            8, 4, 4, 4, 4, 8, 8, 8, 8, 4, 4, 4, 8, 8, 8, 4],
    "am": [1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1],
})

full_model = smf.ols("mpg ~ wt + hp + cyl + am", data=mtcars).fit()

print("Part (a): Full model mpg ~ wt + hp + cyl + am")
print(full_model.summary().tables[1])
print(f"\nThe hp coefficient is {full_model.params['hp']:.4f} (p = {full_model.pvalues['hp']:.4f}).")
print("Holding wt, cyl, and am constant, one additional horsepower is associated "
        "with this many mpg change in the full model.")

vif_design = sm.add_constant(mtcars[["wt", "hp", "cyl"]])
vif_table = pd.DataFrame({
    "predictor": ["wt", "hp", "cyl"],
    "VIF": [variance_inflation_factor(vif_design.values, i) for i in range(1, 4)],
})
print("\nPart (b): VIF values")
print(vif_table.round(3).to_string(index=False))

simpler_model = smf.ols("mpg ~ wt", data=mtcars).fit()
print("\nSimpler model: mpg ~ wt")
print(simpler_model.summary().tables[1])

interaction_model = smf.ols("mpg ~ wt * am", data=mtcars).fit()
print("\nPart (c): Interaction model mpg ~ wt * am")
print(interaction_model.summary().tables[1])
print(f"\nThe wt:am interaction coefficient is {interaction_model.params['wt:am']:.4f} "
        f"(p = {interaction_model.pvalues['wt:am']:.4f}).")

comparison = pd.DataFrame({
    "model": ["Full", "Simpler", "Interaction"],
    "adjusted_R_squared": [full_model.rsquared_adj, simpler_model.rsquared_adj,
                            interaction_model.rsquared_adj],
    "AIC": [full_model.aic, simpler_model.aic, interaction_model.aic],
    "significant_terms_p<0.05": [
        ", ".join(full_model.pvalues[full_model.pvalues < 0.05].index),
        ", ".join(simpler_model.pvalues[simpler_model.pvalues < 0.05].index),
        ", ".join(interaction_model.pvalues[interaction_model.pvalues < 0.05].index),
    ],
})
print("\nPart (d): Model comparison")
print(comparison.round({"adjusted_R_squared": 3, "AIC": 1}).to_string(index=False))

Part (a): Full model mpg ~ wt + hp + cyl + am
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     36.1948      2.969     12.191      0.000      30.103      42.287
wt            -2.5888      0.906     -2.857      0.008      -4.448      -0.730
hp            -0.0236      0.013     -1.789      0.085      -0.051       0.003
cyl           -0.7931      0.528     -1.502      0.145      -1.877       0.291
am             1.5799      1.388      1.138      0.265      -1.268       4.427

The hp coefficient is -0.0236 (p = 0.0848).
Holding wt, cyl, and am constant, one additional horsepower is associated with this many mpg change in the full model.

Part (b): VIF values
predictor   VIF
       wt 2.334
       hp 3.312
      cyl 4.347

Simpler model: mpg ~ wt
                 coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------

### Results and model selection

#### (a) Full model and horsepower

The full model is

$$
\widehat{mpg} = 36.195 - 2.589wt - 0.0236hp - 0.793cyl + 1.580am.
$$

The horsepower coefficient is **-0.0236** ($p=0.0848$). Holding weight, cylinders, and transmission type constant, an increase of one horsepower is associated with an estimated decrease of 0.0236 mpg. This coefficient is not statistically significant at the 5% level.

#### (b) VIF and simpler model

The VIF values are:

| Predictor | VIF |
|---|---:|
| wt | 2.334 |
| hp | 3.312 |
| cyl | 4.347 |

These values do not indicate perfect or severe multicollinearity, although `cyl` shows the strongest overlap with the other predictors. In the full model, `wt` is significant ($p=0.008$), while `hp`, `cyl`, and `am` are not significant at the 5% level. Therefore, a parsimonious model is:

$$
\widehat{mpg} = 37.285 - 5.345wt.
$$

#### (c) Interaction effect

The interaction model is:

$$
\widehat{mpg} = 31.416 - 3.786wt + 14.878am - 5.298(wt\times am).
$$

The interaction coefficient is **-5.298** ($p=0.0010$). This means the weight slope for manual-transmission cars is 5.298 mpg per 1,000 lb more negative than the slope for automatic-transmission cars. The estimated weight slopes are approximately -3.786 for automatic cars and -9.084 for manual cars.

#### (d) Best model

| Model | Adjusted $R^2$ | AIC | Significant non-intercept terms |
|---|---:|---:|---|
| Full | 0.830 | 153.6 | wt |
| Simpler (`wt`) | 0.745 | 164.0 | wt |
| Interaction (`wt * am`) | 0.815 | 155.5 | wt, am, wt:am |

The interaction model is the best overall explanatory choice. It is more parsimonious than the full model, every non-intercept term is statistically significant, and it captures the meaningful fact that the relationship between weight and fuel efficiency differs by transmission type. The full model has the highest adjusted $R^2$ and lowest AIC, so it may be preferred when prediction is the only priority; however, its additional predictors are not individually significant. The interaction model provides a stronger balance of interpretability, parsimony, and statistical significance.

### Question 7 Regression Diagnostics 


### Context: 
- For the final regression model you chose as "best" in Question 6 (the interaction model), 
an economist must verify that the assumptions of OLS are reasonably met before trusting the 
results.

### Question 7: 
-  Perform the following diagnostic checks on your chosen model. For each, state the 
assumption, your method of checking it, your findings, and whether the assumption appears to be 
met. 

- a) Normality of Residuals: Are the model's errors normally distributed?

- b) Homoscedasticity (Constant Error Variance): Do the errors have a constant variance across the 
range of predicted values? 

- c) Linearity: Is the assumption of a linear relationship between the predictors and the outcome 
reasonable? 

- d) Influential Points: Briefly discuss how you would identify highly influential data points and what 
you would consider doing about them. (Conceptual discussion, no new calculations needed). 

In [17]:
from statsmodels.stats.diagnostic import het_breuschpagan, linear_reset
from scipy import stats

# The chosen model from Question 6: mpg ~ wt * am.
residuals = interaction_model.resid
fitted_values = interaction_model.fittedvalues

# (a) Normality of residuals: Shapiro-Wilk test.
shapiro_result = stats.shapiro(residuals)
print("(a) Normality of residuals")
print(f"Shapiro-Wilk statistic: {shapiro_result.statistic:.3f}")
print(f"p-value: {shapiro_result.pvalue:.4f}")

# (b) Homoscedasticity: Breusch-Pagan test.
bp_lm_stat, bp_lm_pvalue, bp_f_stat, bp_f_pvalue = het_breuschpagan(
    residuals,
    interaction_model.model.exog,
)
print("\n(b) Homoscedasticity")
print(f"Breusch-Pagan LM statistic: {bp_lm_stat:.3f}")
print(f"LM p-value: {bp_lm_pvalue:.4f}")
print(f"Breusch-Pagan F statistic: {bp_f_stat:.3f}")
print(f"F-test p-value: {bp_f_pvalue:.4f}")

# (c) Linearity: Ramsey RESET test and residual-versus-fitted pattern.
reset_result = linear_reset(interaction_model, power=2, use_f=True)
residual_fitted_correlation = np.corrcoef(fitted_values, residuals)[0, 1]
print("\n(c) Linearity")
print(f"Ramsey RESET F statistic: {float(reset_result.fvalue):.3f}")
print(f"RESET p-value: {float(reset_result.pvalue):.4f}")
print(f"Correlation between fitted values and residuals: {residual_fitted_correlation:.4f}")

print("\n(d) Influential points")
print("Identify potentially influential observations with leverage, studentized residuals, "
      "\n and Cook's distance. Investigate unusual cars individually before deciding whether "
      "\n to correct data errors, use a justified transformation, fit a robust model, or "
      "\n report a sensitivity analysis. Do not remove observations solely because they are unusual.")

(a) Normality of residuals
Shapiro-Wilk statistic: 0.942
p-value: 0.0872

(b) Homoscedasticity
Breusch-Pagan LM statistic: 1.448
LM p-value: 0.6943
Breusch-Pagan F statistic: 0.442
F-test p-value: 0.7245

(c) Linearity
Ramsey RESET F statistic: 0.019
RESET p-value: 0.8921
Correlation between fitted values and residuals: 0.0000

(d) Influential points
Identify potentially influential observations with leverage, studentized residuals, 
 and Cook's distance. Investigate unusual cars individually before deciding whether 
 to correct data errors, use a justified transformation, fit a robust model, or 
 report a sensitivity analysis. Do not remove observations solely because they are unusual.


### Diagnostic results and conclusions

The diagnostic checks were applied to the interaction model from Question 6, $mpg \sim wt * am$, using a 5% significance level.

#### (a) Normality of residuals

**Assumption:** The model residuals are approximately normally distributed.

**Method and finding:** The Shapiro-Wilk test gave $W=0.942$ and $p=0.0872$. Because $p>0.05$, we fail to reject the null hypothesis of normal residuals. The normality assumption appears reasonable for this sample, although a Q-Q plot would also be useful for checking tail behavior.

#### (b) Homoscedasticity

**Assumption:** The residual variance is constant across the range of fitted mpg values.

**Method and finding:** The Breusch-Pagan test gave an LM statistic of 1.448 with $p=0.6943$ (the equivalent F-test had $p=0.7245$). Since both p-values exceed 0.05, there is not sufficient evidence of heteroscedasticity. The constant-variance assumption appears reasonable.

#### (c) Linearity

**Assumption:** The conditional relationship between the predictors and mpg is correctly represented by the linear model, including the transmission-specific weight slopes.

**Method and finding:** The Ramsey RESET test gave $F=0.019$ and $p=0.8921$. The correlation between fitted values and residuals was approximately 0.0000. These results provide no evidence of important omitted nonlinear structure, so the linearity assumption appears reasonable. A residual-versus-fitted plot would provide an additional visual check.

#### (d) Influential points

Potentially influential observations should be identified using **leverage**, **studentized residuals**, and **Cook's distance**. A high-leverage car has unusual predictor values; a large studentized residual is poorly predicted; and a large Cook's distance means that the observation substantially changes the fitted model. Such observations should be investigated for data-entry errors or unusual but valid cases. They should not be removed merely because they are unusual. Appropriate responses include correcting verified errors, using a justified transformation or robust regression, and reporting a sensitivity analysis with and without the observation.

### Overall assessment

The normality, constant-variance, and linearity checks do not indicate a serious violation of the OLS assumptions for the chosen interaction model. Therefore, the model's coefficient tests and confidence intervals appear reasonably trustworthy, subject to the usual caution about the small sample size and any influential observations.